Update EC2 Instances for NGINX page

madebydna opened the pull request:
### Proposed changes

**Problem**: The "Creating Amazon EC2 Instances" was last updated about 6 years ago and has fallen out of date with the current flow of creating EC2 instances on AWS. The screenshots and some of the links were outdated. The page did also not comply with Hugo formatting standards, and the current style guide. The Hemingway editor's readability score was 8.

**Solution**: 
- followed instructions in the guide to create an AWS EC2 instance and install NGINX Open Source on it using Ansible
- updated instructions and screenshots to match the current AWS EC2 creation flow
- reduced the number of screenshots
- made sure to use Hugo formatting for links, and screenshots
- improved the readability score to a 6

<img width="1160" alt="Screenshot 2025-02-10 at 7 22 45 PM" src="https://github.com/user-attachments/assets/380667a3-0d14-417b-bcb3-f2841654fff6" />

**Testing**: Followed the instructions in the guide to make sure they work and reflect the current AWS UI flow. 

**Please focus on** (optional): 

1. I placed the images in `static/img/` as instructed in the [Contributing docs](https://github.com/nginx/documentation/blob/main/CONTRIBUTING_DOCS.md#how-to-add-images). Should I remove the old screenshots linked in the original version of the article?
2. I removed the section about other configuration management tools like Chef and Puppet since the links were broken and I could only find the articles on the Korean version of the F5 site (e.g. [Installing NGINX and NGINX Plus with Chef](https://www.f5.com/ko_kr/company/blog/nginx/installing-nginx-nginx-plus-chef)). I can add the section back if linking to that site is okay, though.

Closes #99 

### Checklist

Before merging a pull request, run through this checklist and mark each as complete.

- [x] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [x] I have signed the [F5 Contributor License Agreement (CLA)](https://github.com/f5/.github/blob/main/CLA/cla-markdown.md)
- [x] I have ensured that documentation content adheres to [the style guide](/templates/style-guide.md)
- [x] If the change involves potentially sensitive changes, I have assessed the possible impact
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](/README.md) and [`CHANGELOG.md`](/CHANGELOG.md))
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

ADubhlaoich reviewed (approved):
Changes generally LGTM: I may do a second pass in the interim period waiting for another TW to take a look.

With regard to your questions:

1. Yeah, might as well just delete the old ones.
2. I think leaving out the section is fine. We haven't localised the documentation, so having a section with broken links whose only substitute is not in English hurts the overall flow of the document.

Since you're looking at general improvements beyond correcting text, the archetypes I migrated into the repository might be of interest:

https://raw.githubusercontent.com/nginx/documentation/refs/heads/main/archetypes/default.md

We have some standardized page-level information architecture we're slowly applying to pages as we work on them.

ADubhlaoich commented on content/nginx/deployment-guides/amazon-web-services/ec2-instances-for-nginx.md:
```diff
@@ -4,181 +4,139 @@ description: Create Amazon Elastic Compute Cloud (EC2) instances for running NGI
+To install and configure NGINX Open Source or NGINX Plus on an instance, open a terminal window and connect to the instance over SSH.
 
 1. Navigate to the **Instances** tab on the EC2 Dashboard if you are not there already.
+2. Click the row for an instance to select it.
```
If you change all of the explicit numbers to `1.`, Hugo will automatically enumerate them. It's a useful convention, since it means you can change and re-order sequences without being forced to change the numbers every time.

madebydna replied:
Oh, that's nifty! I will update. 🙂

madebydna commented:
@ADubhlaoich I deleted the unused images and updated the list styles to take advantage of Hugo's automatic enumeration. 

Regarding the default archetype, I wasn't sure if for instance the metadata in the archetype represents a new default, or if I should override it with the specific settings in this article. For instance, the archetype has `toc: false` while this article has it set to true. Let me know if I should make some additional changes to bring the article more in accordance with the archetype.

ADubhlaoich commented:
The archetype shows the new "defaults" in terms of what parameters each document should have and what the precise ordering is.

```
description: Create Amazon Elastic Compute Cloud (EC2) instances for running NGINX
  Open Source and F5 NGINX Plus.
docs: DOCS-444
doctypes:
- task
title: Create Amazon EC2 Instances for NGINX Open Source and NGINX Plus
toc: true
weight: 600
```

The only strictly necessary fields are *title* and *weight*: the former is the page title, while the latter affects the ordering of pages or folders that the page is part of.

- *description* is used for SEO metadata: it's not actually visible within documents, but might appear as part of results in a search engine
- *docs* is used as part of a cataloguing system. Every document has a 1:1 relationship with a *docs* code whose vale is formatted as `DOCS-<some-number>`.
- *doctypes* is completely deprecated and was never used.
- *toc* determines if a document will generate a table of contents or not - 'false' by default. It's included in the archetypes for the sake of letting someone writing a document be aware it's an option.

For this document, you'd re-order the fields, throw the *doctypes* line into oblivion and add *type: how-to* and *product: NGINX+*.

The *product* value is determined by where the document lives, not what the subject focus is for the text. The distinction is important, 'cause you'll sometimes see information related to one project inside the documentation for another.

madebydna commented:
@ADubhlaoich thanks for the detailed explanation and guidance. I updated the metadata to match the new default archetype.
